### ISSUE TITLE
Clustering: Breaks marker anchoring for markers not of the "pin" style.

### ISSUE THREAD

**Issue by mpfulc** [googlemaps/android-maps-compose#286]
The new Clustering() implementation appears to have a preset marker image anchor offset.  Or, to be more precise, the composable passed into the clusterItemContent parameter appears to be wrapped in an offset composable when placed on the screen.  I see no way to configure that anchor offset, which appears to be Offset(0.5f, 1.0f), the default assigned to markers.  This does not break "pin" style markers, but breaks all other marker image types as they are not able to be centered on the point.  Adding anchor configurability may be the most robust fix.

**Comment by wangela** [comment]
If you would like to upvote the priority of this issue, please comment below or react with :+1: so we can see what is popular when we triage.
--------

@mpfulc Thank you for opening this issue. 🙏
Please check out these other resources that might help you get to a resolution in the meantime:

- [Check the issue tracker](https://developers.google.com/maps/support/#issue_tracker) - bugs and feature requests for Google Maps Platform APIs and SDKs
- [Open a support case](https://developers.google.com/maps/support#contact-maps-support) - Get 1:1 support in Cloud Console.
- [Discord](https://discord.gg/fHTDmZe9En) - chat with other developers
- [StackOverflow](http://stackoverflow.com/questions/tagged/google-maps) - use the `google-maps` tag


_This is an automated message, feel free to ignore._

**Comment by cwsiteplan** [comment]
any update on this?
seems quite crucial to have that configurability - otherwise clustering cannot be used with custom pins. :(

**Comment by noelb** [comment]
+1 to getting this fixed. Being able to change the content of the marker but not the anchor point really limits the usefulness of it.

**Comment by cwsiteplan** [comment]
any update? this would be a highly appreciated - this is blocker for our product launch.

**Comment by cwsiteplan** [comment]
Thanks for the workaround - will try to implement that.

Still I would see the ability of "setting anchor" as essential - having a Map SDK that does not support custom marker content (and as content varies the anchor needs to be adjusted) seems too limited.

**Comment by jackthebeast** [comment]
@cwsiteplan did you had any luck implementing this? mind sharing your solution?

**Comment by cwsiteplan** [comment]
i subclassed `ClusterRenderer`.

I do set the anchor differently depending on the type of marker and also depending if it's a single item or a cluster.

created a custom composable that takes in a anchorConfig:
```
public fun <T : ClusterItem> rememberZoomClusterRenderer(
    clusterContent: @Composable ((Cluster<T>) -> Unit)?,
    clusterItemContent: @Composable ((T) -> Unit)?,
    clusterManager: ZoomClusterManager<T>?,
    markerAnchorConfig: MarkerAnchorConfig? = null,
    animationEnabled: Boolean = true,
): ClusterRenderer<T>? {
```

this is just basically holding the customized ClusterRenderer

```
public class ZoomClusterRenderer<T : ClusterItem>(
    ...
    private val markerAnchorConfig: MarkerAnchorConfig? = null,
) : ComposeUiClusterRenderer<T>(
    ...
) {
    ...
    override fun onBeforeClusterRendered(cluster: Cluster<T>, markerOptions: MarkerOptions) {
        // set anchor if present
        markerAnchorConfig?.getAnchorOffsetFor(cluster.items.first(), true)?.let {
            markerOptions.anchor(it.x, it.y)
        }
        super.onBeforeClusterRendered(cluster, markerOptions)
    }

    override fun onBeforeClusterItemRendered(item: T, markerOptions: MarkerOptions) {
        // set anchor if present
        markerAnchorConfig?.getAnchorOffsetFor(item, false)?.let { offset ->
            markerOptions.anchor(offset.x, offset.y)
        }

        super.onBeforeClusterItemRendered(item, markerOptions)
    }
}

**Comment by jackthebeast** [comment]
@cwsiteplan  thank you, but how did you extended `ComposeUiClusterRenderer`? seems like most of the classes and functions needed are private/internal. did you imported them all in your project?

**Comment by cwsiteplan** [comment]
yes, unfortunately i need to have the library as "local" module.

**Comment by kikoso** [comment]
Hello. I just realised this issue was still open. This has been released in version [8.1.0](https://github.com/googlemaps/android-maps-compose/releases/tag/v8.1.0). Closing this issue.
